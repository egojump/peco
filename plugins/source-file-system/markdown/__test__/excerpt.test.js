const markdown = require('markdown-it')
const plugin = require('../excerpt')

test('auto', () => {
  const md = markdown()
  const env = {}
  md.use(plugin)
  const html = md.render(
    `
# hello

hello world

oops
  `,
    env
  )
  expect(env).toMatchSnapshot('env')
  expect(html).toMatchSnapshot('html')
})

test('manually', () => {
  const md = markdown({
    html: true
  })
  const env = {}
  md.use(plugin)
  const html = md.render(
    `
# hello
<!-- more -->
hello world

<strong>hehe</strong>

<!-- foo -->

oops
  `,
    env
  )
  expect(env).toMatchSnapshot('env')
  expect(html).toMatchSnapshot('html')
})

test('paragraph only: false', () => {
  const md = markdown()
  const env = {}
  md.use(plugin, {
    paragraphOnly: false
  })
  const html = md.render(
    `
# hello

hello world

oops
  `,
    env
  )
  expect(env).toMatchSnapshot('env')
  expect(html).toMatchSnapshot('html')
})
