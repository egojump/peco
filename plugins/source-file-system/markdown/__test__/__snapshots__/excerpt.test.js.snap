// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auto: env 1`] = `
Object {
  "excerpt": "<p>hello world</p>
",
  "excerpted": true,
  "manualExcerpt": false,
}
`;

exports[`auto: html 1`] = `
"<h1>hello</h1>
<p>hello world</p>
<p>oops</p>
"
`;

exports[`manually: env 1`] = `
Object {
  "excerpt": "<h1>hello</h1>
",
  "excerpted": true,
  "manualExcerpt": true,
}
`;

exports[`manually: html 1`] = `
"<h1>hello</h1>
<!-- more -->
<p>hello world</p>
<p><strong>hehe</strong></p>
<!-- foo -->
<p>oops</p>
"
`;

exports[`paragraph only: false: env 1`] = `
Object {
  "excerpt": "<h1>hello</h1>
<p>hello world</p>
",
  "excerpted": true,
  "manualExcerpt": false,
}
`;

exports[`paragraph only: false: html 1`] = `
"<h1>hello</h1>
<p>hello world</p>
<p>oops</p>
"
`;
